### ISSUE TITLE
Fixing NaNs with built-in Sigmoid

### ISSUE THREAD

**Issue by sergeyf** [maciejkula/triplet_recommendations_keras#1]
Howdy. I found that using the BPR loss from your code as-is led to a slew of NaNs after a couple of epochs. I modified it as follows to use the built-in sigmoid, and it seems to have fixed it:

```
def bpr_triplet_loss(X):

    user_latent, item_latent = X.values()
    positive_item_latent, negative_item_latent = item_latent.values()

    # BPR loss
    loss = -K.sigmoid(K.sum(user_latent * positive_item_latent, axis=-1, keepdims=True)
                                - K.sum(user_latent * negative_item_latent, axis=-1, keepdims=True))

    return loss
```

Thanks for the nice example code!

**Comment by maciejkula** [comment]
Fantastic, thanks a bunch. Would you like to make a PR that I could merge? This would involve three steps:
- change the notebook
- change the `.py` file
- run `nbconvert` on the notebook to update readme.md.

If not I can change that myself!

**Comment by sergeyf** [comment]
Ha, if you're offering: feel free to change it yourself!

**Comment by maciejkula** [comment]
My generosity will be the death of me https://github.com/maciejkula/triplet_recommendations_keras/pull/2

**Comment by sergeyf** [comment]
But what a way to go!
